### PR TITLE
PT-5220 Setting temp file to output_dir in download unpack utils feature

### PR DESCRIPTION
**Pull request by andher1802**
(insert scope and description here)

Items:
* [x] Implemented (new) unit tests
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [x] Bumped version
* [x] Added changelog

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "up42-py"
-version = "1.0.0"
+version = "1.0.1"
```
Comment (javidq) [inline pyproject.toml:3]: This is not a releasing PR, please add postfix

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,11 @@ You can check your current version with the following command:
     ```
 
 For more information, see [UP42 Python package description](https://pypi.org/project/up42-py/).
+## 1.0.1
+
+**May 6, 2024**
+- Set the temp dir to `output_dir` in `download_from_gcs_unpack`.
```
Comment (javidq) [inline CHANGELOG.md:35]: It's a bugfix, we need to add information about the bug that was fixed, not just the changes applied

File: up42/utils.py [hunk 3]
```diff
@@ -111,7 +112,7 @@ def download_from_gcs_unpack(
             directory.
     """
     # Download
-    out_temp = tempfile.mkstemp()[1]
+    out_temp = tempfile.mkstemp(dir=output_directory)[1]
```
Comment (javidq) [inline up42/utils.py:115]: Coverage is missing

Comment (javidq) [inline up42/utils.py:115]: Let's switch to temporary file with context management

**Review by javidq — CHANGES_REQUESTED** [review]

File: up42/utils.py (hunk 3 above)
Comment (andher1802) [reply up42/utils.py]: I had to rename one test that was misnamed. I added tests for the previously missing function and a new one to check that the temp file was in the output_directory.

File: up42/utils.py [hunk 4]
```diff
@@ -109,10 +115,10 @@ def download_from_gcs_unpack(
         download_url: The signed gcs url to download.
         output_directory: The file output directory, defaults to the current working
             directory.
+        delete_compressed_file: Sets if the compressed file should be removed after unpack.
     """
     # Download
-    out_temp = tempfile.mkstemp()[1]
-    with open(out_temp, "wb") as dst:
+    with tempfile.NamedTemporaryFile(dir=output_directory, delete=False) as dst:
```
Comment (javidq) [inline up42/utils.py:121]: Why is delete=False? We want to delete the temp file in the end

File: up42/utils.py [hunk 5]
```diff
@@ -100,6 +105,7 @@ def wrapper(*args, **kwargs):
 def download_from_gcs_unpack(
     download_url: str,
     output_directory: Union[str, pathlib.Path],
+    delete_compressed_file: bool = True,
```
Comment (javidq) [inline up42/utils.py:108]: Not needed

File: up42/utils.py [hunk 6]
```diff
@@ -153,6 +159,8 @@ def download_from_gcs_unpack(
         output_directory,
         [p.name for p in out_filepaths],
     )
+    if delete_compressed_file:
+        os.remove(dst.name)
```
Comment (javidq) [inline up42/utils.py:163]: Not needed

File: CHANGELOG.md [hunk 7]
```diff
@@ -29,6 +29,13 @@ You can check your current version with the following command:
     ```
 
 For more information, see [UP42 Python package description](https://pypi.org/project/up42-py/).
+## 1.0.1a1
+
+**May 6, 2024**
+- Fixed `download_from_gcs_unpack` tempdir set now to `output_dir` previously set to default temp folder
+- Renaming misnamed test for `download_from_gcs_unpack` previously named as `test_download_gcs_not_unpack`
+- Adding missing test coverage for `test_download_gcs_not_unpack`
```
Comment (javidq) [inline CHANGELOG.md:37]: test_download_gcs_not_unpack -> download_gcs_not_unpack

File: tests/test_utils.py [hunk 8]
```diff
@@ -187,79 +187,123 @@ def test_fc_to_query_geometry_multipolygon_raises():
...
+        cloud_storage_url = "http://clouddownload.api.com/abcdef"
+
+        out_tgz = pathlib.Path(__file__).resolve().parent / "mock_data/result_tif.tgz"
+        with open(out_tgz, "rb") as src_tgz:
+            out_tgz_file = src_tgz.read()
+        requests_mock.get(
+            url=cloud_storage_url,
+            content=out_tgz_file,
         )
+        with tempfile.TemporaryDirectory() as tempdir:
```
Comment (javidq) [inline tests/test_utils.py:201]: Builtin tmp_path fixture to be used instead

File: tests/test_utils.py [hunk 9]
```diff
@@ -187,79 +187,123 @@ def test_fc_to_query_geometry_multipolygon_raises():
...
-        content=out_tgz_file,
-    )
-    with tempfile.TemporaryDirectory() as tempdir:
-        with open(pathlib.Path(tempdir) / "dummy.txt", "w", encoding="utf-8") as fp:
-            fp.write("dummy")
-        out_files = utils.download_from_gcs_unpack(
-            download_url=cloud_storage_url,
-            output_directory=tempdir,
+class TestDownloadGcsUnpack:
+    def test_download_gcs_unpack_tgz(self, requests_mock):
```
Comment (javidq) [inline tests/test_utils.py:191]: Name to be changes to refkect what behaviour is tested. The same below

File: tests/test_utils.py [hunk 10]
```diff
@@ -187,79 +187,123 @@ def test_fc_to_query_geometry_multipolygon_raises():
...
+        cloud_storage_url = "http://clouddownload.api.com/abcdef"
```
Comment (javidq) [inline tests/test_utils.py:192]: To be moved to class variable

File: tests/test_utils.py [hunk 11]
```diff
@@ -187,79 +187,123 @@ def test_fc_to_query_geometry_multipolygon_raises():
...
+
+        out_tgz = pathlib.Path(__file__).resolve().parent / "mock_data/result_tif.tgz"
```
Comment (javidq) [inline tests/test_utils.py:194]: pathlib.Path("tests/mock_data/result_tif.tgz") and the similar below

File: tests/test_utils.py [hunk 12]
```diff
@@ -187,79 +187,123 @@ def test_fc_to_query_geometry_multipolygon_raises():
...
+        with open(out_tgz, "rb") as src_tgz:
+            out_tgz_file = src_tgz.read()
+        requests_mock.get(
+            url=cloud_storage_url,
+            content=out_tgz_file,
```
Comment (javidq) [inline tests/test_utils.py:199]: out_tgz.read_bytes()

File: tests/test_utils.py [hunk 13]
```diff
@@ -187,79 +187,123 @@ def test_fc_to_query_geometry_multipolygon_raises():
...
+            with open(pathlib.Path(tempdir) / "dummy.txt", "w", encoding="utf-8") as fp:
+                fp.write("dummy")
+            with pytest.raises(ValueError):
+                utils.download_from_gcs_unpack(
+                    download_url=cloud_storage_url,
+                    output_directory=tempdir,
+                )
+
+
+class TestDownloadGcsNotUnpack:
```
Comment (javidq) [inline tests/test_utils.py:282]: Let's not touch this if we don't provide full coverage here

**Review by javidq — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 7 above)
Comment (andher1802) [reply CHANGELOG.md:37]: Remove these tests as we do not provide full coverage yet. I will create a separate ticket for that.

**Pull request merged**